src/gs-screenshot-image.c

```diff
@@ -214,7 +280,26 @@ gs_screenshot_image_complete_cb (SoupSession *session,
 	if (msg->status_code == SOUP_STATUS_CANCELLED || ssimg->session == NULL)
 		return;
 
+	if (msg->status_code == SOUP_STATUS_NOT_MODIFIED) {
+		SoupURI *uri = soup_message_get_uri (msg);
+		g_autofree char *url = soup_uri_to_string (uri, FALSE);
+
+		g_debug ("Screenshot for '%s' has not been modified; not "
+			 "downloading again", url);
+		return;
+	}
 	if (msg->status_code != SOUP_STATUS_OK) {
+		SoupURI *uri = soup_message_get_uri (msg);
+		g_autofree char *url = soup_uri_to_string (uri, FALSE);
+		g_warning ("Result of downloading screenshot from "
+			   "'%s': %s", url,
+			   soup_status_get_phrase (msg->status_code));
+
+		/* if we are already showing a screenshot, do not show and
+		 * error */
+		if (ssimg->status == GS_SCREENSHOT_IMAGE_STATUS_IMAGE)
+			return;
```

Comment:
It may be worth to wrap this check inside `gs_screenshot_image_set_error()` itself? Or in another function that calls into it?